nomos-sdp/src/lib.rs

```diff
@@ -51,12 +52,71 @@ pub enum ServiceType {
     ExecutorNetwork,
 }
 
+// Add Display and FromStr implementations so we can serialize
```

Comment:
@bacv. @danielSanchezQ I could not make ServiceType and ProviderId serialize nicely from config inside the HashMap<ServiceType...> or HashMap<ProviderId...>, getting message that key must be a string.

I know we already talked about this in the first PR and this solution was rejected, but I added it in this draft as an illustration and wanted to see if you guys have an idea how can I make it work with existing code (transparent serde)

Reply:
uhm, deriving serde is not enough? I do not like that display is implemented for this purpose. That kind of thags are not really descriptive.  Instead probably we want a wrapper type on the hasmaps where we can control the serialization/deserialization when we send it to storage (if that is the problem as It wasnt clear to me).

Reply:
Tbh I thought it was enough as well. But what happens is that serde would use derive when serializing ServiceType, but will not use it when serializing it inside the HashMap. I found this relevant question on stackoverflow:https://stackoverflow.com/questions/68579809/using-serde-to-deserialize-a-hashmap-with-a-enum-key. 

Wrapper type on Hashmaps could be a good solution. I can also try some serde_with combination.

Reply:
Yeah, just the easiest thing to do would do.